internal/slicer/slicer.go

```diff
@@ -158,13 +158,28 @@ func Run(options *RunOptions) (*Report, error) {
 			Package:   slice.Package,
 			Extract:   extract[slice.Package],
 			TargetDir: targetDir,
-			Globbed:   globbedPaths,
 			// Creates the filesystem entry and adds it to the report.
-			Create: func(o *fsutil.CreateOptions) error {
+			Create: func(extractInfo *deb.ExtractInfo, o *fsutil.CreateOptions) error {
 				entry, err := fsutil.Create(o)
 				if err != nil {
 					return err
 				}
+
+				// We only want to keep the entries that were explicitly listed
+				// in the slice definition.
+				if !isListedInSlice(extractInfo, slice) {
```

Comment:
There's an implicit check going on inside this: if `extractInfo` is `nil`, this returns `false` and we are returning from the function. I think the `extractInfo == nil` should be explicit here. So, something like below:

```suggestion
				if extractInfo == nil {
					return nil
				}

				// We only want to keep the entries that were explicitly listed
				// in the slice definition.
				if !isListedInSlice(extractInfo, slice) {
```

Reply:
My idea was to move the check and the for loop to avoid having a very big closure, I can move it back if you think being more explicit means it is more readable.

Reply:
Hmm.. having the `nil` check outside of the `isListedInSlice` and explicitly inside the closure seems better to me still.

Reply:
Done, I was being silly. I could access the map without a for loop.